### PR TITLE
Completed all but the debugging challenge

### PR DESCRIPTION
**Pull request by NikkiDunlap**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ X ] Coding Challenges
- [ X ] Interview Questions
- [ ] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

I did not complete the debugging challenge.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -51,17 +104,51 @@ const studyMinutesWeek2 = {
...
 
 // b) Create the function that makes the test pass.
+
+const accumulatedSum = (array) => {
+  let sum = 0
+  return array.map = (number) => {
+    sum += number
+    return sum
+  }
+}
```
Comment (DFWilliams) [inline code-challenges.test.js:154]: Nice work on your coding challenges Nikki! One thing to note, if at first your pseudo code fails and you find a new solution make sure to update your pseudo to  reflect your new solution.

File: react-debugging.html [hunk 2]
```diff
@@ -81,6 +83,7 @@ <h1>Draw a Card</h1>
               )
             })}
           </div>
+          </>
```
Comment (DFWilliams) [inline react-debugging.html:86]: Great effort! Take a look at  lines 71 and 74, what could be the cause of bugs and how can you resolve it?

File: interview-questions.md [hunk 3]
```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Your answer:
+Your answer: A React component is similar to a puzzle piece. It helps the website operate the way that the developer intends for it to operate. There are function and class components. Functional components are simple and hold a small amount of data; however, class components are more complex and change over time. 
 
-Researched answer:
+Researched answer: A React component is a building block that can be recognized as a JavaScript function that returns a React element. There are function components that are simple and concise, and there are class components that can hold state.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: The state values represents the internal data of a component, like the user input. It usually contains the data that will continue to change over time. It is like a variable. Props are similar to arguments that are passed into a component. They cannot be changed within the component.
```
Comment (DFWilliams) [inline interview-questions.md:29]: This is a good explanation, I would encourage to use examples whenever you can get the chance. Examples are your best friend!

File: interview-questions.md [hunk 4]
```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: Props are can be used to pass data from one component to another; however, they cannot be modified. State values can change; each component has a state.
 
 5. What is the DOM?
 
-Your answer:
+Your answer: The DOM (Document Object Model) is an interface that allows us to manipulate the HTML properties of a web page. 
 
-Researched answer:
+Researched answer: The DOM is a tree-like structured representation of all of the HTML elements on a web-page or application.
```
Comment (DFWilliams) [inline interview-questions.md:37]: This would be a great place to add more content! Like do you know of any libraries or frameworks that use the DOM?

**Review by DFWilliams — APPROVED** [review]
Nikki this is some good stuff! Your code is very clean and you continue to display an Adv. Beginner level on comprehension. I left some comments on your coding challenges and your interview question that I feel will help you continue to grow. I want to encourage you to dig deeper on your interview questions. A great way to show that you know and understand the concept give examples. Everybody loves a goodNikki this is some good stuff! Your code is very clean and you continue to display an Adv. Beginner level on comprehension. I left some comments on your coding challenges and your interview question that I feel will help you continue to grow. I want to encourage you to dig deeper on your interview questions. A great way to show that you know and understand the concept give examples. Everybody loves a good example!  Great job this week, keep it up  example!  Great job this week, keep it up

File: code-challenges.test.js (hunk 1 above)
Comment (NikkiDunlap) [reply code-challenges.test.js:154]: Thank you! I always forget to change my pseudocode when I find a new way to work the problem. I'll definitely remember that!

File: react-debugging.html (hunk 2 above)
Comment (NikkiDunlap) [reply react-debugging.html:86]: I am missing my function. The function will produce a random card every time the button is clicked. In the function, we'll create an array that includes all of the suits and their values. To ensure that  each click produces a random card, we'd use Math.floor & math.random. Then we could use string interpolation to inform the user of what card they've pulled.

File: interview-questions.md (hunk 3 above)
Comment (NikkiDunlap) [reply interview-questions.md:29]: Thank you! I am definitely going to start doing this. It gives me better understanding too.

File: interview-questions.md (hunk 4 above)
Comment (NikkiDunlap) [reply interview-questions.md:37]: I definitely should have gone more in detail

**Pull request merged**